pytest_mh/utils/firewall.py

```diff
@@ -289,95 +302,77 @@ def _next_priority(self) -> int:
-        :param ip_family: If the rules is added as IPv4, IPv6 rule or both, defaults to ``all``.
-        :type ip_family: Literal["ipv4", "ipv6", "all"], optional
-        :param priority: Rule priority, defaults to None (= auto-assign next value)
-        :type priority: int | None, optional
-        :return: Rule priority, to be used for rule removal.
+        Add a new (permanent) policy.
+
+        Except for the name, all parameters are optional. When the priority is not provided,
+        the next priority is assigned. When the other parameters is not provided,
+        no value is assigned to the newly created policy and it defaults to ``firewalld``'s
+        defaults.
+
+        :param ingress: The ingress zone, not assigned if not provided.
+        :type ingress: str | None, optional.
+        :param egress: The egress zone, not assigned if not provided.
+        :type egress: str | None, optional.
+        :param priority: Rule priority, defaults to the next priority.
+        :type priority: int | None, optional.
+        :param target: Rule target, not assigned if not provided.
+        :type target: Literal["CONTINUE", "ACCEPT", "DROP", "REJECT"] | None, optional.
+        :return: Policy priority.
         :rtype: int
         """
+        self.logger.info(f'Firewalld: adding policy "{name}"')
+
+        cmd = ["firewall-cmd", "--permanent", "--new-policy", name]
+        self.host.ssh.exec(cmd, log_level=SSHLog.Error)
+        cmd[2] = "--policy"
+
         if priority is None:
             priority = self._next_priority
+        self.host.ssh.exec([*cmd, "--set-priority", str(priority)], log_level=SSHLog.Error)
 
-        cmd = [table, chain, priority, *args]
+        if ingress is not None:
+            self.host.ssh.exec([*cmd, "--add-ingress-zone", ingress], log_level=SSHLog.Error)
 
-        if ip_family in ["ipv4", "all"]:
-            self.logger.info(f'Firewalld: adding IPv4 direct firewall rule: {" ".join([str(x) for x in cmd])}')
-            self.host.ssh.exec(["firewall-cmd", "--direct", "--add-rule", "ipv4", *cmd], log_level=SSHLog.Error)
+        if egress is not None:
+            self.host.ssh.exec([*cmd, "--add-egress-zone", egress], log_level=SSHLog.Error)
 
-        if ip_family in ["ipv6", "all"]:
-            self.logger.info(f'Firewalld: adding IPv6 direct firewall rule: {" ".join([str(x) for x in cmd])}')
-            self.host.ssh.exec(["firewall-cmd", "--direct", "--add-rule", "ipv6", *cmd], log_level=SSHLog.Error)
+        if target is not None:
+            self.host.ssh.exec([*cmd, "--set-target", target], log_level=SSHLog.Error)
 
         return priority
 
-    def remove_direct_rule(
-        self,
-        priority: int,
-        chain: str,
-        args: list[Any],
-        *,
-        table: str = "filter",
-        ip_family: Literal["ipv4", "ipv6", "all"],
-    ) -> None:
-        """
-        Remove direct rule.
+    def remove_policy(self, name: str) -> None:
```

Comment:
I know it's a simple method but I'd expect some comment explaining what it does and the parameters.

Reply:
Done